### DISCUSSION TITLE
simpler approach alarm panel to USR IOT (serial over IP) no IP150?

### DISCUSSION THREAD

**Discussion by jpieren** [ParadoxAlarmInterface/pai#467] [Ideas]
Hi,

I am contemplating using a serial cable from the MG5050 to an USR IOT single RS232 device. This converts the stream to TCP/Telnet. I would assume that the voltage of the MG serial is not an issue for the USR IOT device. I would use the TX/RX and GND.
https://www.amazon.com/Ethernet-Converter-Support-Upgraded-USR-TCP232-302/dp/B083RCHRTL
https://www.pusr.com/products/1-port-rs232-to-ethernet-converters-usr-tcp232-302.html
or might be better with https://shop.usriot.com/ethernet-to-serial-converters-usr-tcp232-306.html (US$26)

cable + 230pcs 2.54mm Pitch Terminal Kit / Housing/ Pin Header Wire Connector Adaptor + DB9 plus IOT device.
  
What do you think about this? There is no soldering needed for this aproach which might appeal to some.

cheers Jay

**Comment by DrMaboul** [comment]
While it sounds as a clean and good idea, I'm not sure it'll work.

As far as I know, the protocols are similar, 1 bit at a time, available bauds rate, parity, stop...

However voltage ranges differ : 
- TTL voltage range is from 0v for logical low level (0) to Vcc (typically 3.3v or 5v) for logical high level (1). Paradox TTL use 5v from what I've read in the wiki.
- RS232 use negative -3v to -15v for logical high level (1) and similar positive voltage for logical low level (0). I.e. on a computer it is -13v to +13v so I assume that the mentioned above adapters will use -13v/+13v.

At the very least, this would require an RS232 to TTL 5v adapter like this one [https://www.serialcomm.com/serial_rs232_converters/rs232_rs485_to_ttl_converters/rs232_to_5v_ttl_converter/rs232_to_5v_ttl.product_general_info.aspx](https://www.serialcomm.com/serial_rs232_converters/rs232_rs485_to_ttl_converters/rs232_to_5v_ttl_converter/rs232_to_5v_ttl.product_general_info.aspx)

Regards

**Reply by jpieren** [reply]
Thank you for the explanation. By the sounds of this it seems Paradox is using RS485. Looking at the USR products they have two that support RS485 to Ethernet:  USR-TCP232-304 or USR-TCP232-306. Seems also to make the wiring easier (no rs232 connector needed).

I will give it a try with the https://www.pusr.com/products/1-port-rs485-to-ethernet-converters-usr-tcp232-304
at Amazon $27.

regards

**Reply by yozik04** [reply]
Something like this worked well enough with EVO192. 

Just found this amazing item on AliExpress. Check it out! €1.82  71％ Off | 3pcs/lot =1PCS PL2303HX+1PCS CP2102+1PCS CH340G USB TO TTL for arduino PL2303 CP2102 5PIN USB to UART TTL Module
https://a.aliexpress.com/_EHNHvIT

**Reply by DrMaboul** [reply]
@jpieren I don't think you'll go further with this solution for PAI. The serial conenctor on the panel do not use RS485 but TTL. I know there a modules for using RS485 to deport serial connection "far way", the CV4USB, but again it needs an additional module except if the MG5050, which I don't own or know, has a RS485 build-in interface.

@yozik04 So cheap ! I'd like to try it with a cheap USB over ethernet extender like this [https://www.amazon.com/Extender-Extension-Compatible-Computers-Keyboards/dp/B0CM3KXBTB](https://www.amazon.com/Extender-Extension-Compatible-Computers-Keyboards/dp/B0CM3KXBTB)

**Reply by jpieren** [reply]
@DrMaboul  thank you for clarifying on TTL (learned something new). 

@yozik04 good idea with the 3pcs/lot =1PCS PL2303HX+1PCS CP2102+1PCS CH340G USB TO TTL for arduino PL2303 CP2102 5PIN USB to UART TTL Module and the USB extender. if it works for EVO should be fine the MG5050.

thank you guys, appreciated!

**Comment by jpieren** [comment]
Hi,

I just noted that my board is a MG5050+. It has two serial ports and according to the manual one is RS485. Both work with 307B serial cable to my  laptop. I just wonder if a RS485 to Ethernet IoT device will do the job to connect. In the mean time I bought the device [usr-tcp232-304](https://www.pusr.com/products/1-port-rs485-to-ethernet-converters-usr-tcp232-304) but I wodner about the pin layout 4 pins on the board. The 304 has RS485 T+/T-/R+/R- how would that connect to the board? Top two from the board Rx and bottom to Tx?

cheers
Jay

**Reply by yozik04** [reply]
https://github.com/ParadoxAlarmInterface/pai/wiki/Connection-methods#serial-connection

**Reply by DrMaboul** [reply]
Where do you see that one of the port is a RS485 on the panel ?

I don't own a MG5050+ but an EVOHD+ (same new generation with two serial ports). The two are serial port and shouldn't work simultaneously. Not tested but following installation guide.
None of them is RS485.

**Comment by jpieren** [comment]
Thank you saw that:

Not sure how that should connect:
RS485 on the IOT____| (RS485 on Panel)
(Pin 1) A+                   <-|-> PIn 2 Tx?
(Pin 2) B-                   <-|-> Pin 1 Rx?
(Pin 3) R+                  <-|->
(Pin 4) R-                  <-|->

(Pin 5) 5-36V DC+- <-|-> 
(Pin 6) 5-36V DC-   <-|-> Pin 3 GND?? 

And Panel Aux we leave off?

any help that understands more of electrical circuit than me is welcome.

kind regards
Jay

**Reply by yozik04** [reply]
A and B is usually modbus. It is not what is expected.

**Reply by jpieren** [reply]
then connecting to Pin3/4 instead A/B is what you are suggesting? What about GND from the MG5050+ panel to Pin6 as mentioned?

**Reply by DrMaboul** [reply]
Serial ports on the MG5050+ are TTL type serial ports and won't work with anything else except using adapters/converters.
See upper link from yozik04 for the pinout (TTL RX 5v, TTL TX 5x, GROUND, AUX +12V)

```
┌───────┐
│ Rx   ┌╵
│ Tx   │
│ GND  │
│ AUX+ └╷
└───────┘

```

And to finish, as far as I know, MG5050+ will suffer of the native encryption design of the serial port on new "+" products (like MG505+,EVOHD+,...).
Easy to verify: using a keypad, go to section 300 for SP/MG or 3000 for EVOxx. If the value is "SERIAL LOCKED", then serial communication is encrypted and only PARADOX modules are compatible. You can disable encryption by buying an unlock code by an official PARADOX reseller to change this and open the use of serial bus.

**Reply by jpieren** [reply]
@DrMaboul 
I understand that the serial port is TTL, however the MG5050+ manual states the second port being **RS485 on page 9 circle 6**.
[MG5000+Installation Guide.pdf](https://github.com/user-attachments/files/16419297/MG5000%2BInstallation.Guide.pdf)
My problem is, I have only K10H/V pad no screen to check section 300. Is there a way from Babyware to check that? The MG5050+ has V1.20.001.

**Reply by DrMaboul** [reply]
@jpieren About the RS485 port, I don't see it in the manual I have which is only about MG5050+ and not described as draft on the cover of the book. This one doesn't have any reference to RS485 anymore.
![image](https://github.com/user-attachments/assets/e35932bd-96d7-4a53-988e-97f7b899b499)


I'm afraid that all new model (ending with "+") are defaulted encrypted.

As far as I know, it's impossible to use Babyware to look at section values.
The manual says TM50/70, but any keyboard with LCD will do (K32, K641, K656). Seems difficult to look at values with a K10. Not an expert.
